consensus/spos/bn/subroundEndRound.go

```diff
@@ -5,10 +5,21 @@ import (
 	"time"
 
 	"github.com/ElrondNetwork/elrond-go/consensus/spos"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/statusHandler"
 )
 
 type subroundEndRound struct {
 	*spos.Subround
+
+	appStatusHandler core.AppStatusHandler
+}
+
+// SetAppStatusHandler method set appStatusHandler
+func (sr *subroundEndRound) SetAppStatusHandler(handler core.AppStatusHandler) {
```

Comment:
The pattern already used in app status handlers impl is that this func should return error if the provided handler is nil. Let's keep this pattern.

Reply:
SetAppStatusHandler return error now